packages/open-truss/src/configuration/engine-v1/config-schemas.tsx

```diff
@@ -28,6 +31,47 @@ workflow:
 const ViewPropsV1Shape = z.record(z.string(), YamlShape).optional()
 export type ViewPropsV1 = z.infer<typeof ViewPropsV1Shape>
 
+const ViewV1Shape = z
+  .object({
+    component: z.string(),
+    props: ViewPropsV1Shape,
+  })
+  .superRefine((view, ctx) => {
+    const COMPONENTS = RUNTIME_COMPONENTS()
+    const componentName = view.component
+    const Component = COMPONENTS[componentName]
+
+    if (Component === undefined) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: `Did not find ${componentName}`,
+        fatal: true,
+      })
+      return z.NEVER
+    }
+
+    if (view.props && hasPropsExport(Component)) {
+      const expectedComponentProps = describeZod(Component.Props.shape)
+      Object.entries(view.props).forEach(([propName, propValue]) => {
+        const expectedProp = expectedComponentProps[propName]
+        if (expectedProp === undefined) {
+          // TODO probably fine to allow props not declared by the component?
+          return
+        }
+
+        const errors = isPropValueValid(expectedProp, propValue)
```

Comment:
I'm a little surprised we need this -- `isPropValueValid` looks a lot like reimplementing zod's validation system. Can you say more about why you did it this way versus zod's validations, which I think would be `Component.Props.parse(view.props)`?